doc/src/running-suites.rst

```diff
@@ -0,0 +1,1634 @@
+order of events - which can vary according to the external queueing
+conditions, runahead limit, and so on.
+
+To prepare a reference log for a suite, run it with the
+``--reference-log`` option, and manually verify the
+correctness of the reference run.
+
+To reference test a suite, just run it (in dummy mode for the most
+comprehensive test without running real tasks) with the
+``--reference-test`` option.
+
+A battery of automated reference tests is used to test cylc before
+posting a new release version. Reference tests can also be used to check that
+a cylc upgrade will not break your own complex
+suites - the triggering check will catch any bug that causes a task to
+run when it shouldn't, for instance; even in a dummy mode reference
+test the full task job script (sans ``script`` items) executes on the
+proper task host by the proper batch system.
+
+Reference tests can be configured with the following settings:
+
+.. code-block:: cylc
+
+   [cylc]
+       [[reference test]]
+           suite shutdown event handler = cylc check-triggering
+           required run mode = dummy
+           allow task failures = False
+           live mode suite timeout = PT5M
+           dummy mode suite timeout = PT2M
+           simulation mode suite timeout = PT2M
+
+
+Roll-your-own Reference Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the default reference test is not sufficient for your needs, firstly
+note that you can override the default shutdown event handler, and
+secondly that the ``--reference-test`` option is merely a short
+cut to the following suite.rc settings which can also be set manually if
+you wish:
+
+.. code-block:: cylc
+
+   [cylc]
+       abort if any task fails = True
+       [[events]]
+           shutdown handler = cylc check-triggering
+           timeout = PT5M
+           abort if shutdown handler fails = True
+           abort on timeout = True
+
+
+.. _SuiteStatePolling:
+
+Triggering Off Of Tasks In Other Suites
+---------------------------------------
+
+.. note::
+
+   Please read :ref:`External Triggers` before using
+   the older inter-suite triggering mechanism described in this section.
+
+The ``cylc suite-state`` command interrogates suite run databases. It
+has a polling mode that waits for a given task in the target suite to achieve a
+given state, or receive a given message. This can be used to make task
+scripting wait for a remote task to succeed (for example).
+
+Automatic suite-state polling tasks can be defined with in the graph. They get
+automatically-generated task scripting that uses ``cylc suite-state``
+appropriately (it is an error to give your own ``script`` item for these
+tasks).
+
+Here's how to trigger a task ``bar`` off a task ``foo`` in
+a remote suite called ``other.suite``:
+
+.. code-block:: cylc
+
+   [scheduling]
+       [[dependencies]]
+           [[[T00, T12]]]
+               graph = "my-foo<other.suite::foo> => bar"
```

Comment:
Inter-suite triggers render in red as errors, my fault again, the `cylc_lang` lexer doesn't like these (probably the colon character). Again these errors can be ignored in this PR.

I've updated the table in #2752 accordingly.

Reply:
No problem, thanks for investigating. I did notice there were a few syntax highlighting abnormalities in the built docs & intended to add that into one of the lists for future work (& seem to have neglected to). But #2752 is a more appropriate home.

Reply:
I've now fixed these issues in https://github.com/metomi/rose/pull/2274, we can copy across the updated `cylc_lang.py` once it's merged (in a future PR if this gets merged first).